engine/common/rpc/convert/execution_data.go

```diff
@@ -320,10 +320,11 @@ func MessageToRegisterID(m *entities.RegisterID) (flow.RegisterID, error) {
 	if m == nil {
 		return flow.RegisterID{}, ErrEmptyMessage
 	}
-	return flow.RegisterID{
-		Owner: m.GetOwner(),
-		Key:   m.GetKey(),
-	}, nil
+
+	owner := string(m.GetOwner())
+	key := string(m.GetKey())
+
+	return flow.NewRegisterID(owner, key), nil
```

Comment:
If `owner` has to be Valid UTF8, and `NewRegisterID` is taking a string as `owner`, then it's possible `owner` is not a valid string, such as `hello`. In that case, it should return error. 

I wonder where did we do such validation?

I would also suggest to let `flow.NewRegisterID` take `owner` as `Address` type, instead of a string, which forces the caller to validate the owner string, and only call NewRegisterID with valid Address type. But there is an edge case, where the two service level keys have owner as empty, I think we could just define a zero address for the service level keys.

Reply:
> I wonder where did we do such validation?

This is done natively in the protobuf code, so the client could would fail to generate the request payload.

> I would also suggest to let flow.NewRegisterID take owner as Address type, instead of a string

https://github.com/onflow/flow-go/pull/5144